src/Framework/TestCase.php

```diff
@@ -97,8 +97,13 @@ public function runTest($method, array $args = NULL)
 				if (!is_array($res) && !$res instanceof \Traversable) {
 					throw new TestCaseException("Data provider $provider() doesn't return array or Traversable.");
 				}
-				foreach ($res as $set) {
-					$data[] = is_string(key($set)) ? array_merge($defaultParams, $set) : $set;
+				foreach ($res as $setName => $set) {
+					$set = is_string(key($set)) ? array_merge($defaultParams, $set) : $set;
+					if (is_string($setName)) {
+						$data[$setName] = $set;
+					} else {
+						$data[] = $set;
+					}
```

Comment:
Yes it can. However the probability of that is lower then for regular array. The same string name must be specified explicitly however same array index is quite hard to find where the problem is.